src/daft-table/src/lib.rs

```diff
@@ -43,7 +43,7 @@ impl Table {
         let mut num_rows = 1;
 
         for (field, series) in schema.fields.values().zip(columns.iter()) {
-            if field != series.field() {
+            if field.name != series.field().name || field.dtype != series.field().dtype {
```

Comment:
I had to relax this because we instantiate Fields pretty naively in a lot of Series code without passing along metadata information (e.g. `Series::empty(name, dtype)`). This means that the metadata field on the Series does tend to drift from the schema at some point.

I don't think this is a problem specifically with propagating field_ids, but any metadata on the inferred arrow2 Field likely would have caused issues here as well.

This feels like a reasonable change to me I think? But someone else should chime in too.

Reply:
I ended up doing a custom `PartialEq` implementation on `Field` instead